### PR TITLE
Using main branch for tools/workflow in CD

### PR DESCRIPTION
**Pull request by khanhntd**
**Description:** <Describe what has changed.>

Instead of using the committed cha branch, it would be better for us to use the tools/workflow from the main branch since it is up to date with the github workflow.

### CODE REVIEW

File: .github/workflows/CD.yml [hunk 1]
```diff
@@ -74,7 +76,7 @@ jobs:
       - name: Cache packages
         uses: actions/cache@v2
         with:
-          key: "${{ env.PACKAGE_CACHE_KEY }}"
+          key: "${{ env.TESTING_FRAMEWORK_REPO }}"
```
Comment (jefchien) [inline .github/workflows/CD.yml:79]: Why was this changed? The cached build is meant to be the release candidate.

File: .github/workflows/CD.yml [hunk 2]
```diff
@@ -315,14 +317,24 @@ jobs:
...
           path: adot
+
+      #Since the tools from workflow are always uptodate and to keep it up with the github workflow, we need to check out the main branch
+      #And use the tools from there instead of the commited sha branch to avoid commit additional code for the commited sha's branch
+      #Documentation on side by side checkout: https://github.com/actions/checkout#checkout-multiple-repos-side-by-side
+
+      - name: Checkout main branch to get workflow tools
+        uses: actions/checkout@v2
+        with:
+          ref:  "${{ env.MAIN_BRANCH }}"
```
Comment (jefchien) [inline .github/workflows/CD.yml:336]: I don't think you need to make this an environment variable.

File: .github/workflows/CD.yml [hunk 3]
```diff
@@ -315,14 +317,24 @@ jobs:
...
+          path: main
```
Comment (jefchien) [inline .github/workflows/CD.yml:337]: This also doesn't need to be in a separate path, since the input SHA checkout is already under the `adot` path, but it's fine.

File: .github/workflows/CD.yml [hunk 4]
```diff
@@ -340,7 +352,7 @@ jobs:
         id: version
         run: |
           TAG=`cat build/packages/VERSION`
-          TARGET_VERSION=$TAG bash adot/tools/workflow/docker-version-compare.sh
+          TARGET_VERSION=$TAG bash main/tools/workflow/docker-version-compare.sh
```
Comment (jefchien) [inline .github/workflows/CD.yml:355]: If you're going to change this to use the main branch instead of the input SHA, you should also change the path for the other tools being used.

File: .github/workflows/CD.yml (hunk 3 above)
Comment (khanhntd) [reply .github/workflows/CD.yml:337]: True. Intend to do the nested checkout but it would clear this way.

File: .github/workflows/CD.yml (hunk 2 above)
Comment (khanhntd) [reply .github/workflows/CD.yml:336]: Reasonable. Fixed it.

File: .github/workflows/CD.yml (hunk 4 above)
Comment (khanhntd) [reply .github/workflows/CD.yml:354]: The only used tool with the path ```adot``` is the releasing of dot operator. Other steps are using cache with build/packages. So I think it should be fine.

File: .github/workflows/CD.yml [hunk 5]
```diff
@@ -124,7 +125,7 @@ jobs:
       - name: Restore cached packages
         uses: actions/cache@v2
         with:
-          key: "${{ env.PACKAGE_CACHE_KEY }}"
+          key: "${{ env.TESTING_FRAMEWORK_REPO }}"
```
Comment (jefchien) [inline .github/workflows/CD.yml:128]: Again. Why is this the testing framework repo? Why are you changing the cache keys?

File: .github/workflows/CD.yml (hunk 4 above)
Comment (jefchien) [reply .github/workflows/CD.yml:354]: That is not true. On line 370, the image mirror is using the `adot` path.

File: .github/workflows/CD.yml [hunk 6]
```diff
@@ -315,14 +316,24 @@ jobs:
...
```
Comment (jefchien) [inline .github/workflows/CD.yml:319]: This one as well.

File: .github/workflows/CD.yml (hunk 5 above)
Comment (khanhntd) [reply .github/workflows/CD.yml]: Another fault from me. Changed it back to the original. Should check the repo commit before going somewhere else.

File: .github/workflows/CD.yml (hunk 6 above)
Comment (khanhntd) [reply .github/workflows/CD.yml]: This one is my fault as well. Changed it back to original.

File: .github/workflows/CD.yml (hunk 4 above)
Comment (khanhntd) [reply .github/workflows/CD.yml:354]: Yeah you are right. I misunderstood the use of the image mirror. Fixed it.

Comment (khanhntd) [reply .github/workflows/CD.yml:354]: Also if that's the case, then we don't need to checkout for the input SHA anymore. Nice catch.

File: .github/workflows/CD.yml [hunk 7]
```diff
@@ -33,6 +33,7 @@ env:
   TF_VAR_aws_secret_access_key: ${{ secrets.INTEG_TEST_AWS_KEY_SECRET }}
   TF_VAR_aoc_vpc_name: aoc-vpc-large
   TF_VAR_aoc_vpc_security_group: aoc-vpc-security-group-large
+  TESTING_FRAMEWORK_REPO: 'aws-observability/aws-otel-collector-test-framework'
```
Comment (sethAmazon) [inline .github/workflows/CD.yml:36]: Good clean up

**Review by sethAmazon — APPROVED** [review]

File: .github/workflows/CD.yml [hunk 8]
```diff
@@ -318,11 +319,12 @@ jobs:
...
 
-      - name: Checkout the sha
+      #Since the tools from workflow are always up-to-date and to keep it up with the github workflow, we need to check out the main branch
+      #And use the tools from there instead of the commited sha branch to avoid commit additional code for the commited sha's branch
+      - name: Checkout main branch to get workflow tools
         uses: actions/checkout@v2
         with:
-          ref: ${{ github.event.inputs.sha }}
-          path: adot
+          ref: "main"
```
Comment (jefchien) [inline .github/workflows/CD.yml:327]: This is not required. If we trigger the workflow dispatch, it'll use whatever branch is selected, which should be `main`, but can be another branch.

Comment (khanhntd) [reply .github/workflows/CD.yml]: Reasonable to me after thinking. Fixed it.

**Review by jefchien — APPROVED** [review]
LGTM
